### DISCUSSION TITLE
メモ内容の受け渡しについて

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#15] [Question-original] [answered]
Memo型のmemoを配列で宣言することができれば、それ以外を配列で宣言することなく、メモの内容を受渡できそうと考えたのですがどのように書けば良いでしょうか？
MemoViewControllerの32行目を変えれば配列で表現できそうだと思い色々試してみたのですが、なかなかうまく行かないので教えて頂きたいです。

**Comment by txxxxc** [comment]
お疲れさまです！質問ありがとうございます！
質問の仕方について一つお願いがあります🙏

お手数ですが以下のフォーマットで質問の内容をまとめ直して頂けるとありがたいです！
```
(1) 実現したいこと
(2) 実現するために自分が試した内容とその結果
(3) 調べた内容や資料
(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
(5) GitHubのリンク
```

理由としては、自分が渡した情報としゅんひぐが欲しい情報の相違を無くすためです。
現在頂いている質問の内容的にこちらが質問を答える上で足りない情報が多いので、ご協力してもらえると助かります！

最初から質問のフォーマットを共有すれば良かったです！ごめんね🙇‍♂️
お手数ですが、よろしくお願いします！

**Reply by ShunHigu** [reply]
すいません！！やり直します！
（１）MemoViewControllerの33行目のmemoを配列で宣言したいです。
配列で表現することができたらRealmを用いて、たとえばmemo[1]の中にToDo、詳細、期限の内容を保存することが出来て、TableViewのCellを押した際（prepare()でindexPathの数を引き継ぐ）に内容が表示されると考えます。
さらに、memo[1]のToDoの内容をprepareでTableViewのページに引き継ぐことでCellのタイトルにすることでき検索機能を追加しやすくなると考えました。
（２）MemoViewControllerの33行目を
           let memo:[String]:Memo?=read()
          と書き直して試してみたり、
          let memo:[String]=[]
          memo:Memo?=read() 
    　  と書き直してみたりしました。
（３）今回はMemoという型を用意しているので、その時の配列の表現の仕方を調べてみたのですが、ただの配列の宣言の仕方の説明ばかりで載っている記事がなかったです。
（４）MemoViewControllerの33行目です。
（５）https://github.com/ShunHigu/TodoList
よろしくお願いしいます！

自分のRealmの仕組みへの理解が甘いので、そもそものやろうとしていること自体が間違っていましたら教えていただきたいです。

**Reply by txxxxc** [reply]
> 配列で表現することができたらRealmを用いて、たとえばmemo[1]の中にToDo、詳細、期限の内容を保存することが出来て、TableViewのCellを押した際（prepare()でindexPathの数を引き継ぐ）に内容が表示されると考えます。

もちろん配列を表現することでもtodoの更新を実現することはできますが、memo単体を保持するだけでデータの更新機能は実現できるのでその方法をおすすめします！
データの更新方法に関してはこれを参考にしてもらえると幸いです！↓
https://naoya-ono.com/swift/realm-update-delete/

具体的に言うと、現在MemoViewControllerで保持しているiの代わりにtodoをLobbyViewControllerから渡してあげてそれを更新すると言う流れです！

他に不明な点があれば教えてもらえると🙏

**Reply by ShunHigu** [reply]
ありがとうございます！！

**Reply by txxxxc** [reply]
解決したら教えてくださいー！

**Reply by ShunHigu** [reply]
LobbyViewControllerでどのようにしてtodoの内容を渡すということは、
LobbyViewControllerから渡されたiを用いたtitleArray[i]がToDoの内容なので、MemoViewControllerの36行目でRealmの中でtitleとtitleArray[i]の内容が同じものを検索して、39行目で表示するというやり方であっていますか？

**Comment by txxxxc** [comment, marked as answer]
あ、流れとしてはこんな感じです！
1. LobbyViewControllerにてRealmを使ってデータベースからTodoを全件取得し、保持する
2. 押されたCellのindexを元にTodoを配列から取り出す
3. 取り出したTodoをMemoViewControllerに渡して、表示させる

1を既に実装済みかと思ってた！ごめん🙏
あと、行数を伝える時は行数に実際のコード(と周辺の+-5行程度)を添えてもらえると嬉しいです！！！

**Reply by txxxxc** [reply]
> それを配列とイコールにすることで内容を配列から取り出すことが出来るということですか？

配列とイコールにするというか、Realmのデータにアクセスすると配列(厳密にいうと配列のようなもの)が返ってきます。

例として、MemoViewControllerに記載されているread()で説明してみると、
```
    func read() -> Memo?{
        return realm.objects(Memo.self).first
    }
```

1. まず、`realm.objects(Memo.self)`の部分ではRealmに保存されているMemoという型のデータを**全件**取得しています。
2. そして`first`の部分で返ったきた配列の一番最初の要素を取り出してます！
なので、`realm.objects(Memo.self)`というふうにすれば配列を取り出すことができます！

これを踏まえてもう一回実装してみてほしいですー！エラーが出ている部分は実装する過程で消えたりすると思うので思うので、もう一回分けて質問してもらえると！！！

**Reply by ShunHigu** [reply]
ありがとうございます！

すいません！
reply出来ていないと思い先程の投稿消してしまいました！Slackに文の内容は残っているので確認できるので大丈夫です！
